src/Processor/Braintree.php

```diff
@@ -207,4 +212,9 @@ public function findPlanById(string $planId): ?Plan
     {
         return $this->plans->find($planId);
     }
+
+    public function makeSubscriptionQuery(): SubscriptionQuery
+    {
+        return clone $this->subscriptionQuery;
```

Comment:
Why not just `return new SubscriptionQuery($this->braintree, $this->subscriptionMapper)`? 

This does require that `braintree` and `subscriptionMapper` be placed on `$this`, but that seems better than cloning this jam every time. (especially if the query has state on it)

Reply:
That's cool with me... I was assuming we were keeping braintree and the mapper out of the class for a reason, so went for a solution that would leave that as is.

Updated.